File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.js
function solution(s) {
    let ret = s.length;
    
    for(let len = 1;len<=s.length/2;len++){
        let str = "", next = "";
        let cnt = 1;
        let cur = s.substr(0,len);
        // console.log(cur);
        let i;
        for(i=len;i<s.length-len+1;){
            next = s.substr(i,len);
            // console.log(next);
            if(cur == next) {
                // console.log("TT");
                cnt++;
            }
            else {
                if(cnt>1)str += cnt;
                str += cur;
                // console.log(str);
                cnt = 1, cur =next;
            }
            i+=len;
        }
        if(cnt>1) str+=cnt;
        str+= next + s.substr(i);
        
        // console.log(str);
        if(ret > str.length){
            ret = str.length;
        }
    }
    return ret;
}

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.cpp
#include <string>
#include <vector>

using namespace std;

vector<string> convert(string s,int siz){
    vector<string> v;
    for(int i=0;i<s.size();i+=siz){
        v.push_back(s.substr(i,siz));
    }
    
    return v;
}
int solution(string s) {
    int n = s.size();
    int ret = s.size();
    
    for(int siz=n/2; siz>=1;siz--){
        vector<string> v = convert(s,siz);
        string pre = v[0], str = "";
        int cnt = 1;
        
        for(int i=1;i<v.size();i++){
            string now =v[i] ;
            if(pre==now) cnt++;
            else {
                if(cnt != 1) str+= to_string(cnt);
                str += pre;
                pre = now;
                cnt = 1;
            }
        }
        
        if(cnt != 1) str+= to_string(cnt);
        str += pre;
        ret = (ret>str.size() ? str.size(): ret);
        
    }
    return ret;
}